src/modalities/dataloader/create_packed_data.py

```diff
@@ -1,31 +1,37 @@
+import math
+import multiprocessing
+import os
 import pickle
 import warnings
 from pathlib import Path
-from typing import IO
+from typing import Callable, List, Tuple
 
 import jq
-import numpy as np
 from tqdm import tqdm
 from transformers import PreTrainedTokenizer
 
 from modalities.dataloader.large_file_lines_reader import LargeFileLinesReader
 
 
+class EmptySampleError(RuntimeError):
+    pass
+
+
 class PackedDataGenerator:
-    # amount of bytes to represent tokens as integers.
-    # If the vocabulary exceeds 2^(8*`size_in_bytes`), this requires adaptation.
-    TOKEN_SIZE_IN_BYTES = 4
     # amount of bytes to represent number of all tokens in dataset.
     # If the amount exceeds 2^(8*`header_size_in_bytes`), this requires adaptation.
     # Decided to keep this constant, since a size of 8 bytes requires more data than the internet currently provides
-    HEAD_SIZE_IN_BYTES = 8
+    DATA_SECTION_LENGTH_IN_BYTES = 8
+    TOKEN_SIZE_DESCRIPTOR_LENGTH_IN_BYTES = 4
+    HEADER_SIZE_IN_BYTES = DATA_SECTION_LENGTH_IN_BYTES + TOKEN_SIZE_DESCRIPTOR_LENGTH_IN_BYTES
 
     def __init__(
         self,
         src_path: Path,
         tokenizer: PreTrainedTokenizer,
         index_path: Path = None,
         jq_pattern: str = ".text",
+        number_of_processes: int = 60,
```

Comment:
For the entrypoint this derived automatically for the number of CPUs. But yeah fixing something here still doesn't make sense. I ll just let it use the number of CPUs as well. Otherwise constructors calls grow too much